nvm.sh

```diff
@@ -909,13 +909,30 @@ nvm_get_os() {
 }
 
 nvm_get_arch() {
-  local NVM_UNAME
-  NVM_UNAME="$(uname -m)"
+  local HOST_ARCH
+  local NVM_OS
+
+  NVM_OS="$(nvm_get_os)"
+  # If the OS is SunOS, first try to use pkgsrc to guess
+  # the most appropriate arch. If it's not available, use
+  # isainfo to get the instruction set supported by the
+  # kernel.
+  if [ "_$NVM_OS" = "_sunos" ]; then
+    HOST_ARCH=$(pkg_info -Q MACHINE_ARCH pkg_install)
```

Comment:
Right, good catch :+1: The original intent was to catch this case with:
```
if [ $EXIT_CODE -ne 0 ]; then
  HOST_ARCH=$(isainfo -n)
fi
```
but the `local` command before the test for `$?` would overwrite `$?`. So I moved the `local` declaration at the top of the function and left it as is. I also [added tests to catch that problem](https://github.com/creationix/nvm/pull/854/files#diff-7a60576df1bb49d940a58f93de5a1fc7R72). 

I think having a `nvm_has_pkg_info` function is a bit overkill for this use case, but If you'd still prefer to handle it that way I'm fine with it too.